### ISSUE TITLE
Implement custom training loop using gradient tapes 

### ISSUE THREAD

**Issue by galacticglum** [GalacticGlum/composer#2]
This is so that non-Keras models can be used.

Currently, the ``train`` command in ``cli.py`` is rather limited by the fact that it calls ``model.fit``.
